proxyd/cache.go

```diff
@@ -75,6 +81,29 @@ func (c *redisCache) Put(ctx context.Context, key string, value string) error {
 	return err
 }
 
+func (c *redisCache) Clear(ctx context.Context) error {
+	patterns := []string{"lvc:*", "method:*"}
+
+	for _, p := range patterns {
```

Comment:
This could take a while to run; delaying bootup. In the heyday we'd have over 100k `method:*` keys cached.
Perhaps we can lazily delete `method:` keys by checking its expiration at lookup time. Or inspect the cached data to figure out if it part of a reorg'd chain. For example, using the block number/hash in the `eth_getBlockByNumber`. Hopefully there's a solution we can use that doesn't complicate caching logic.

Reply:
If we set expirations on all keys, then we can increment a "generation" number embedded in the key. Redis can then evict keys based on their expiration/LRU if memory gets low.

Reply:
Another advantage there is that you won't need to modify `patterns` above if there are additional key prefixes stored in Redis.

Reply:
Cool idea. One caveat though is multiple proxyd instances writing to the same cache will may conflict whenever a new generation is warranted (i.e. when consensus is broken and they all clear the cache at the same time).
One solution is to have a "current generation key" that can be atomically incremented via `INCR` to coordinate writes across proxyd instances.

Reply:
I'll leave the actual cache invalidation logic out of this PR and merge only the event emitter for now.